### PR TITLE
Remove mentions of IRC in favor of the new Contributors Chat platform

### PR DESCRIPTION
**Pull request by pycbouh**
I couldn't help myself and adjusted some texts around the parts I've edited, so if a version for `3.2` is desired, I can make a separate PR.

### CODE REVIEW

File: about/introduction.rst [hunk 1]
```diff
@@ -80,7 +80,7 @@ unbalanced distribution of contents – but the way it is split up should be
 relatively intuitive:
 
 - The :ref:`sec-general` section contains this introduction as well as
-  information about the engine, its history, its licensing, authors, etc. It
+  the information about the engine, its history, its licensing, authors, etc. It
```
Comment (Calinou) [inline about/introduction.rst:83]: I think the previous form was correct already. "information" is often used in a non-numerable form.

File: about/introduction.rst [hunk 2]
```diff
@@ -90,13 +90,13 @@ relatively intuitive:
 - The :ref:`sec-tutorials` section can be read as needed,
   in any order. It contains feature-specific tutorials and documentation.
 - The :ref:`sec-devel` section is intended for advanced users and contributors
-  to the engine development, with information on compiling the engine,
-  developing C++ modules or editor plugins.
-- The :ref:`sec-community` section gives information related to contributing to
-  engine development and the life of its community, e.g. how to report bugs,
+  to the engine development, with the information on compiling the engine,
```
Comment (Calinou) [inline about/introduction.rst:93]: Same as above.

File: about/introduction.rst [hunk 3]
```diff
@@ -90,13 +90,13 @@ relatively intuitive:
...
+  contributing to the editor, or developing C++ modules.
+- The :ref:`sec-community` section gives the information related to contributing to
```
Comment (Calinou) [inline about/introduction.rst:95]: Same as above.

File: about/introduction.rst [hunk 4]
```diff
@@ -90,13 +90,13 @@ relatively intuitive:
...
+  the engine development and the life of its community, e.g. how to report bugs,
```
Comment (Calinou) [inline about/introduction.rst:96]: Same as above, I think "engine development" on its own feels right.

File: community/asset_library/submitting_to_assetlib.rst [hunk 5]
```diff
@@ -161,14 +161,18 @@ submission boxes to enable them.
...
-be patient! You will be informed when your asset is reviewed. If it was rejected,
+Once you are done, press "Submit". Your asset will be entered into the review queue.
+You can check all assets currently pending a review `here <https://godotengine.org/asset-library/asset/edit?&asset=-1>`_ .
+The approval process is manual and may take up to a few days for your asset to be accepted (or rejected), so please
+be patient! 
+
+.. note::
+  You may have some luck accelerating the approval process by messaging the
+  moderators and AssetLib reviewers on the `Contributors Chat <https://chat.godotengine.org/>`_,
+  or the official Discord server.
```
Comment (Calinou) [inline community/asset_library/submitting_to_assetlib.rst:172]:
```suggestion
.. note::

    You may have some luck accelerating the approval process by messaging the
    moderators and AssetLib reviewers on the `Contributors Chat <https://chat.godotengine.org/>`_,
    or the official Discord server.
```

File: community/channels.rst [hunk 6]
```diff
@@ -14,9 +14,17 @@ Q&A
...
+Rocket.Chat
+-----------
+
+- `Godot Contributors Chat <https://chat.godotengine.org/>`_
+
 IRC on Freenode
 ---------------
 
+.. note::
+  As of January 2021, core developer chat has moved to the Godot Contributors Chat platform listed above.
```
Comment (Calinou) [inline community/channels.rst:26]:
```suggestion
.. note::

    As of January 2021, core developer chat has moved to the Godot Contributors Chat platform listed above.
```

File: about/introduction.rst (hunk 1 above)
Comment (pycbouh) [reply about/introduction.rst:83]: I've specifically double-checked this before adjusting, and found out that if you refer to a specific information, which we do in the cases I've adjusted, you can use the article.

File: community/asset_library/submitting_to_assetlib.rst (hunk 5 above)
Comment (pycbouh) [reply community/asset_library/submitting_to_assetlib.rst:172]: Actually, we don't use that consistently at all, and half the time we start the text on the same line as `.. note::` and indent to that same level, but sure 🙂

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**